File: questaoLogica/quadradoraiz.js
function quadradoraiz() {//criando metodo para encontrar o quadrado do menor numero e a raiz do menor numero
    let numberOne = document.getElementById("numberOne").value, //criando uma variavel numberOne e atribuindo a ela o valor obtido no campo com id numberOne
        numberTwo = document.getElementById("numberTwo").value, //criando uma variavel numberTwo e atribuindo a ela o valor obtido no campo com id numberTwo
        raiz, quadrado; //criando variaveis para receber o valor da raiz e do quadrado

    if (numberOne.length === 0 || numberTwo.length === 0) { //verifico se ambos os inputs estão preenchidos
        alert("Preencha ambos os campos!"); // alerto o usuario se o campo estiver vazio
        return; //encerro a funçao
    }

    if (numberOne > numberTwo) { //verifico se o primeiro numero é maior que o segundo numero
        quadrado = Math.pow(numberTwo, 2); // uso o metodo pow presente na biblioteca Math para calcular a potencia do menor numero
        raiz = numberOne > 0 ?  Math.sqrt(numberOne) : "Impossivel"; // uso o metodo sqrt presente na biblioteca Math para calcular a raiz do maior numero
        //uso uma operaçao ternaria para verificar se o segundo numero é maior que zero, caso sim executo sqrt, caso nao coloco impossivel na varivel raiz 

        alert('O quadrado do menor número é: ' + quadrado + '\n A raiz quadrada do maior número é: ' + raiz); //exibo os resultados
        return; //encerro a funçao
    }

    //caso o segundo numero seja maior ou ambos os numeros sejam iguais essa etapa é executada

    quadrado = Math.pow(numberOne, 2); 
    raiz = numberTwo > 0 ? Math.sqrt(numberTwo) : "Impossivel";

    if (numberTwo === numberOne) { //verifica se são iguais
        alert('Os números são iguais!\n O quadrado do número é: ' + quadrado + '\n A raiz quadrada do número é: ' + raiz); //exibe os resultados 
        return; //encerro a funçao
    }

    //caso o segundo numero seja maior e não seja igual ao primeiro essa etapa é executada

    alert('O quadrado do menor número é: ' + quadrado + '\n A raiz quadrada do maior número é: ' + raiz); // exibo o resultado 

}

File: questaoLogica/quadradoraiz.c
#include <stdio.h> //incluindo a biblioteca com os metodos printf, scanf
#include <stdlib.h> //incluindo a biblioteca padrao para atribuicao e alocacao de memoria
#include <math.h> //incluindo a biblioteca para o calculo da potencia e da raiz

void main()
{
    float vetor[2] = {0,0}, maior = 0; //definido as variaveis do tipo float que irei usar
    int result; //definindo variavel do tipo int
    
    printf("Entre com o primeiro numero:"); //printando texto para o usuario
    result = scanf("%f", &vetor[0]); //obtendo a entrada do teclado e colocando o retorno de exito ou falha do scanf em result 
    
     if (result != 1) { //se o usuario entrar com algo que nao seja um numero o result sera 0, indicando um erro
        printf("Entrada invalida. Certifique-se de inserir um numero.\n");
        return; //encerro o programa
    }

    printf("Entre com o segundo numero:");
    result = scanf("%f", &vetor[1]); 
    
     if (result != 1) {
        printf("Entrada invalida. Certifique-se de inserir um numero.\n");
        return; //encerro o programa
    }
   
   system("clear");//limpo a janela
   
   //caso houvesse uma entrada n de numeros poderiamos ordenar usando bubble sort ou heap sort, é mais eficiente em complexidade de tempo
    if(vetor[0] > vetor[1]){ //ordenando o vetor para que o menor numero seja o primeiro e o maior o ultimo usando substituição 
        maior = vetor[0];
        vetor[0] = vetor[1];
        vetor[1] = maior;
    }
    
    if (vetor[0] == vetor[1])//verifico se os numeros sao iguais
    {
        printf("Ambos os números são iguais!\n");//se forem iguais exibo uma mensagem
    }
    
    printf("Quadrado do menor número(%.4f): %.4f\n", vetor[0], pow(vetor[0], 2));//uso o metodo pow para obter a potencia e exibo o resultado
    
    if(vetor[1] < 0){ //verifico se o maior numero é negatico ou positivo
        printf("Raiz quadrada do maior número(%.4f): impossivel", vetor[1]); // se for negativo informo que é impossivel fazer a raiz
        return; //encerro o programa
    }
    
    printf("Raiz quadrada do maior número(%.4f): %f", vetor[1], sqrt(vetor[1])); //uso o metodo sqrt para obter a raiz e exibo o resultado
}